gnovm/pkg/gnolang/op_call.go

```diff
@@ -58,6 +58,13 @@ func (m *Machine) doOpCall() {
 	clo := fr.Func.GetClosure(m.Store)
 	b := m.Alloc.NewBlock(fr.Func.GetSource(m.Store), clo)
 	m.PushBlock(b)
+	if fv.nativeBody == nil && fv.NativePkg != "" {
```

Comment:
OTOH, if we already attach nativeBody upon preprocess, why not just fail here instead of having two places where we load it?

Reply:
Maybe because (a) preprocessing should check existence anyways, so might as well do it eagerly there and (b) it also makes sense to load lazily in the future after we remove PreprocessAllFilesAndSaveBlockNodes?

Reply:
Because Preprocess is not executed on realm recovery; we don't store `nativeBody` in the realm storage because it's a closure, instead we store NativePkg/NativeName.

So in the case where we load+run (ie. gno test, gno run), this is not needed, but on-chain it definitely is.

Reply:
right ok.